state/api/src/main/java/com/expediagroup/streamplatform/streamregistry/state/EntityView.java

```diff
@@ -25,6 +26,25 @@
 
 /**
  * Provides a unified view of the current state of all entities held in Stream Registry.
+ * <p/>
+ * The underlying state storage from which this is this is loaded may support entities which have been deleted. If
+ * that is the case it is important to understand the lifecycle of entities as it relates to {@link EntityView}.
+ * <p/>
+ * Entities may be created and updated in the underlying state storage. Any entity which has been created (but not
+ * deleted) will be returned by {@link #all(Class)} and {@link #get(Key)}. It is expected that users handle said
+ * entities accordingly (update external dependencies/respond to view requests etc).
```

Comment:
Perhaps simplify it:
```suggestion
 * Entities may be created and updated in the underlying state storage. Any entity which has been created (but not
 * deleted) will be returned by {@link #all(Class)} and {@link #get(Key)}. 
```